### PR TITLE
feat: enable REST query API by default

### LINKED ISSUE #25993: Enable Query API (not configurable, remove dedicated annotation for query controllers)
## Description

* Remove dedicated query controllers from the REST API.
* Existing query controllers are integrated into the regular controllers if available, e.g., the `UserTaskQueryController` is integrated into the `UserTaskController`.
* The flag "camunda.rest.query.enabled" is removed and not considered anymore.
* Configuration, integration, and REST API tests are adjusted accordingly.

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

The REST controllers don't distinguish between regular and query API anymore. All V2 C8 REST endpoints are enabled by default and can only be disabled together.

The configuration option "camunda.rest.query.enabled" will not be considered anymore.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #25993

### CODE REVIEW

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestOperateWithExporter.java [hunk 1]
```diff
@@ -148,7 +148,6 @@ protected SpringApplicationBuilder createSpringBuilder() {
 
     withProperty("zeebe.broker.gateway.enable", true);
     withProperty("camunda.rest.enabled", true);
```
Comment (nathansandi) [inline qa/util/src/main/java/io/camunda/qa/util/cluster/TestOperateWithExporter.java:150]: 🤔 [not in the scope] Should we consider enable this property as true by default in the future?

**Review by nathansandi — COMMENTED** [review]
💯  Great work over here, one thing to check: 

🔨  I think there are a couple of unit tests failing on the branch (on rest-gateway module)
related to the UserTask

File: qa/util/src/main/java/io/camunda/qa/util/cluster/TestOperateWithExporter.java (hunk 1 above)
Comment (tmetzke) [reply qa/util/src/main/java/io/camunda/qa/util/cluster/TestOperateWithExporter.java:150]: It already is, actually, see [default config test](https://github.com/camunda/camunda/blob/88d0b4df0681c15923aa5c51087756bb3e6e00dd/zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/configuration/RestApiDefaultConfigurationTest.java#L35-L47)

**Comment by tmetzke** [comment]
> > 🔨 I think there are a couple of unit tests failing on the branch (on rest-gateway module) related to the UserTask
> 
> Yeah, I need to adjust test mocking mostly, forgot about that.

CI is back to normal 🙂 👍 
@nathansandi, let me know if you spotted anything else, otherwise, let's approve and merge this before new incompatible changes come in from someone 😅

**Review by houssain-barouni — APPROVED** [review]

**Pull request merged**